### DISCUSSION TITLE
課題３についての質問

### DISCUSSION THREAD

**Discussion by TakedaAyano** [nsc-kiwamu/traning-takeda#7] [Q&A]
質問がいくつかございます。

１．「5050」と表示させられたのですが、それはnumberに5050を指定しているからでしょうか？
２．「25」と表示させるには、どこの数字を変えればいいのかわかりませんでした。

全体的に、サンプルを見てもコードの仕組みを私の力では理解することができませんでした。
また、FizzBuzzの問題はネットで調べて、それのマネをして作成しました。

**Comment by nsc-kiwamu** [comment]
## 回答1、2
解決しているようなので、特にコメントはありません。

今はなんとなく動かしたり、ネットのものを流用して動いた。(=がよくわらかん)
というところもあると思います。

私もソースコードを見ていて「ここどうやって動いているんだ？」
と思うことはあります。

そんな時にどう理解するかですが、処理を細分化したり値を変えたりして色々と動かして理解します。
例えば「if (i % 2 !== 1)」ってなんだろう。。。
と思ったとしたら
・細分化
　console.log(i % 2)
　confole.log((1 % 2 !==1))
　のような記載を追加して、何をしているのか見たりします。
それで何をしているのかなんとなくわかったら、今度は値を変えて試してみます
・値を変えて検証
　console.log(i % 3)
　confole.log((1 % 3 !==1))
　2→3に変えることで結果が〇〇になるはずだ！と仮説を立てる
　実際に動かしてみて仮説と合っているか確認する。
　※仮説と合っていれば理解が深まっているし、違っていれば再度値を変えて検証して理解を深めるといった感じです

**Reply by TakedaAyano** [reply]
ご回答ありがとうございます。

細分化し、値を変え、その上で「こうなるはず」という仮説を立てて考えることを意識したいと思います。
